src/features/foods/components/FilterBadge.tsx

```diff
@@ -0,0 +1,44 @@
+import { CrossIcon } from '@/assets/images/icons';
+import { Icon } from '@/components/ui';
+import { Badge } from '@/components/ui/Badge';
+import { cn } from '@/lib/tailwind/utils';
+
+import { ComponentType } from 'react';
+
+interface IFilterBadgeProps {
+  icon?: ComponentType;
+  emoji?: string;
+  text: string;
+  className?: string;
+  onCrossClick?: () => void;
+}
+
+const FilterBadge = ({
+  icon,
+  emoji,
+  text,
+  className,
+  onCrossClick,
+  ...props
+}: IFilterBadgeProps) => {
+  const CircledIcon = icon;
+  return (
+    <Badge variant="light" className={cn('pl-1 pr-0 gap-0', className)} {...props}>
+      {CircledIcon ? (
+        <div className="bg-white w-4 h-4 rounded-full p-[3.5px] mr-1 flex items-center justify-center">
+          <CircledIcon />
+        </div>
+      ) : emoji ? (
+        <div className="bg-white w-4 h-4 rounded-full p-[3px] mr-1 flex items-center justify-center text-2xs select-none">
+          {emoji}
+        </div>
+      ) : null}
+      {text}
+      <button className="h-full w-6 flex items-center relative" onClick={onCrossClick}>
```

Comment:
### (Optional)

<img width="90" alt="Screenshot 2023-12-18 at 11 59 17" src="https://github.com/genesis-tech-tribe/nishiki-frontend/assets/99339182/56ed62ba-3b9e-4032-862f-18a62a64f8ea">

For now, there's this yellow squared ring on focusing(by typing tab to set focus to cross button)
The focus design is not designed in wireframe, but maybe is it better to at least set the ring shape circle now?

Reply:
Thank you for sharing your perspective. We should have talked about that earlier.

I have been thinking of that since I started creating the first shared components, `Button` and `Dialog`. But honestly, I purposely dismissed it because, in mobile, the cursor's focus never moves around using the [ tab ] key, while the input's needs the ring style when focused because it does happen.

As you say, the ring style for buttons and links is not defined in Figma. I would tackle that all together in a different issue/PR when the ring style is required.